### ISSUE TITLE
Question about transcription and translation

### ISSUE THREAD

**Issue by ZJMEI** [uchicago-bio/2016-Autumn-Forum#10]
In problem 5 we are expected to generate a protein sequence from a given DNA sequence. The output contains a "frame_position", I have no idea what it is. Does it means one DNA sequence may contain several protein sequences(frames) after translation? Though I don't think so.

Also, is the given DNA sequence a coding strand or a template strand?

**Comment by jvmakin** [comment]
A sequence of DNA has two strands (i.e. the one you have and its compliment) and three 'open reading frames' a piece - six total. Since amino acid codons are three base pairs in length, if you start reading at frame 1 (i.e. the first base pair) it is equivalent to starting to read at pair 4 (think about i!) and so on for frames 2 and 3. The negative frames refer to the opposite strand.

Its important to know what reading frame you're looking at when you're describing a possible sequence because without this knowledge, the amino acid sequence will come out totally different! 

Wiki has a pretty good explanation:
https://en.wikipedia.org/wiki/Open_reading_frame

Hope this helps!

**Comment by jvmakin** [comment]
http://www.sci.sdsu.edu/~smaloy/MicrobialGenetics/topics/chroms-genes-prots/temp-strand.html

^ This is a good description of coding v. template strands. 
TLDR; The template strand is transcribed into mRNA and the coding strand is identical to said mRNA (with T instead of U). 

For this particular sequence of DNA, we don't know which is which. That is the reason you're looking at all possible reading frames on both strands!

**Comment by ZJMEI** [comment]
@jvmakin Thanks a lot! It really help!

**Comment by ZJMEI** [comment]
Just one more question about what a negative frames really means.

If I have a DNA such like 'CTGAT', we may have two mRNA sequence 'CUGAU'(treat original DNA as a coding strand) and 'GACUA' (as a template strand). So now when I'm considering the first frame of 'CUGAU', should it be frame position 1 or negative 1?

**Comment by jvmakin** [comment]
No problem! 

If you're treating the original DNA as the coding strand, it should be frame position 1. 

Here is a nice graphic that shows it visually:
http://www.oligo.net/tour-analyze-orf.html

This is a great way to check your work (be sure to cite!):
http://www.bioinformatics.org/sms2/orf_find.html

Godspeed.

**Comment by ZJMEI** [comment]
Got it! Thanks again.
